lib/checkother.cpp

```diff
@@ -3554,3 +3554,188 @@ void CheckOther::overlappingWriteFunction(const Token *tok)
+                                break;
+                            }
+                        }
+                    }
+                }
+            }
+
+         
+            if (!error_found) {
+                bool matched = Token::Match(tok, "%var% = %var% ;");
+                if (!matched && Token::Match(tok, "%var% . %var% = %var% ;"))
+                {
+                    tok = tok->tokAt(2);
+                    matched = true;
+                }
+                if (matched)
+                {
+                    const Variable* svar2 = tok->tokAt(2)->variable();
+                    const Variable* svar = tok->variable();
+                    if (svar && !svar->isArgument() && svar2 && svar != svar2)
+                    {
+                        if (!IsSameName(svar->name(), svar2->name()))
+                        {
+                            for (auto const& targ : tmpArgListInfo)
+                            {
+                                if (IsSameName(svar->name(), targ.varname))
+                                {
+                                    if (mSettings->severity.isEnabled(Severity::error) || mSettings->certainty.isEnabled(Certainty::inconclusive))
+                                    {
+                                        for (auto const& targ2 : tmpArgListInfo)
+                                        {
+                                            if (svar2->name() == targ2.varname)
+                                            {
+                                                mismatchingNamesWriteError(targ.tok, svar->name(), svar2->name(), targ.varname);
+                                                error_found = true;
+                                                break;
+                                            }
+                                        }
+                                    }
+                                    if (!error_found)
+                                    {
+                                        if (mSettings->severity.isEnabled(Severity::warning) && !mSettings->certainty.isEnabled(Certainty::inconclusive))
+                                        {
+                                            mismatchingNamesWriteError(targ.tok, svar->name(), svar2->name(), targ.varname);
+                                            error_found = true;
+                                        }
+                                    }
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+void CheckOther::mismatchingNamesWriteError(const Token* tok, std::string var, std::string arg, std::string newname)
+{
+    reportError(tok, mSettings->severity.isEnabled(Severity::error) ? Severity::error : Severity::warning, "mismatchingNamesWriteError", var + " and " + arg + " name mismatch. Did you mean: " + newname);
```

Comment:
we use "error" if there is definitely a bug whenever the code is executed.
we use "warning" if the code is dangerous and can lead to undefined behavior.
this is a "style" warning. there is no undefined behavior just a possible mistake. I would guess this will detect copy/paste mistakes primarily.